backend/profile/profile.controller.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { allPublishedProfilesPayload } from './profile.service'
+
+export async function requestHandler(
+  method: string | undefined,
+  callbackFn: () => Promise<void>,
+) {
+  switch (method) {
+    case 'GET':
+      return callbackFn()
+    case 'POST':
+      return callbackFn()
+    default:
+      throw new Error('Unsupported HTTP method')
+  }
+}
+
+export async function handleGetRequest(
```

Comment:
**nitpick:** We don't need to create controllers this way in next.js app because `api/profiles/route` is already, a controller. We are duplicating the same logic this way. Let's call this logic there, and remove this file.